test/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter_integration_test.h

```diff
@@ -0,0 +1,80 @@
+#include "test/integration/http_integration.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+const std::string ADAPTIVE_CONCURRENCY_CONFIG =
+    R"EOF(
+name: envoy.filters.http.adaptive_concurrency
+config:
+  gradient_controller_config:
+    sample_aggregate_percentile:
+      value: 50
+    concurrency_limit_params:
+      concurrency_update_interval: 0.1s
+    min_rtt_calc_params:
+      interval: 30s
+      request_count: 50
+)EOF";
+
+const std::string CONCURRENCY_LIMIT_GAUGE_NAME =
+    "http.config_test.adaptive_concurrency.gradient_controller.concurrency_limit";
+const std::string REQUEST_BLOCK_COUNTER_NAME =
+    "http.config_test.adaptive_concurrency.gradient_controller.rq_blocked";
+const std::string MIN_RTT_GAUGE_NAME =
+    "http.config_test.adaptive_concurrency.gradient_controller.min_rtt_msecs";
+
+class AdaptiveConcurrencyIntegrationTest
+    : public testing::TestWithParam<Network::Address::IpVersion>,
+      public Event::TestUsingSimulatedTime,
+      public HttpIntegrationTest {
+public:
+  AdaptiveConcurrencyIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, GetParam()) {}
+
+  void customInit() {
+    setDownstreamProtocol(Http::CodecClient::Type::HTTP2);
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+    config_helper_.addFilter(ADAPTIVE_CONCURRENCY_CONFIG);
+    initialize();
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+  }
+
+  void TearDown() override {
+    HttpIntegrationTest::cleanupUpstreamAndDownstream();
+    codec_client_->close();
+    codec_client_.reset();
+  }
+
+protected:
+  // Send some number of requests with 'delay_ms' specifying the amount of time the fault filter
+  // will delay them.
+  void sendRequests(const uint32_t request_count, const uint32_t num_forwarded);
+
+  // Waits for a specified duration and then responds to all queued up requests in a FIFO manner.
+  // Asserts that the expected number of requests are forwarded through the filter. The oldest
+  // requests are the forwarded requests.
+  //
+  // Note: For interleaved forwarded/blocked requests, use respondToRequest() directly.
+  void respondToAllRequests(const int forwarded_count, const std::chrono::milliseconds latency);
```

Comment:
nit: remove const from pass by value params here and elsewhere, it doesn't do anything.